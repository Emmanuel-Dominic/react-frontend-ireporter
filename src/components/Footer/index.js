/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import '../../assets/scss/Footer.scss';


const Footer = () => (
  <React.Fragment>
    <div className="footer-dark">
      <footer>
        <div className="container">
          <div className="row">
            <div className="col-md-3 item text">
              <ul>
                <li><a href="#" className="a-text">About</a></li>
                <li><a href="#" className="a-text">Contact</a></li>
                <li><a href="#" className="a-text">Advertise</a></li>
              </ul>
            </div>
            <div className="col-md-6 item text">
              <h3>
                <span className="head-text">&#8465;&#8477;&#8455;</span>
                PORT
                <span className="head-text">&#8455;&#8477;</span>
              </h3>
              <p>
                Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
                Aenean commodo ligula eget dolor. Aenean massa.
              </p>
            </div>
            <div className="col-md-3 item social">
              <a href="#"><i className="fab fa-facebook-square social-fab" /></a>
              <a href="#"><i className="fab fa-twitter social-fab" /></a>
              <a href="#"><i className="fab fa-instagram social-fab" /></a>
            </div>
          </div>
        </div>
      </footer>
    </div>
  </React.Fragment>
);

export default Footer;
