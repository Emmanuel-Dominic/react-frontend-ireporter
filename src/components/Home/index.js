import React, { Component } from 'react';
import '../../assets/scss/Home.scss';

class Home extends Component {
  render() {
    return (
      <div className="body"><h1 className="home">Welcome to the Home page</h1></div>
    );
  }
}
export default Home;
