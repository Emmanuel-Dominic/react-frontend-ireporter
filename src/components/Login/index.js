/* eslint-disable jsx-a11y/anchor-is-valid */
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import PropTypes from 'prop-types';
import React from 'react';
import loginAction from 'store/actions/Login';

import 'assets/scss/Login.scss';

export class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      detailError: '',
    };
  }

  handleChange = (e) => {
    e.preventDefault();
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    const { email, password } = this.state;
    const credentials = {
      email,
      password,
    };
    const { login } = this.props;

    if (this.validateDataEntry(email, password)) {
      return;
    }
    await login(credentials);
    this.setState({
      email: '',
      password: '',
    });
  }

  validateDataEntry(email, password) {
    if ((email === '') || (password === '')) {
      toast.dismiss();
      this.setState({
        detailError: 'Email & Password fields are Required',
      });
      return true;
    }
    return false;
  }

  render() {
    const { detailError } = this.state;
    return (
      <div className="body">
        <form className="container">
          <h2 className="sr-only text-center">
            <strong className="head-text">Login</strong>
            Form
          </h2>
          <div className="illustration">
            <i className="icon ion-ios-navigate" />
          </div>
          <p className="errors">{detailError}</p>
          <div className="container">
            <div className="form-group">
              <input id="email" type="email" name="email" onChange={this.handleChange} placeholder="Email" className="form-control" />
            </div>
            <div className="form-group">
              <input
                id="password"
                type="password"
                name="password"
                onChange={this.handleChange}
                placeholder="Password"
                className="form-control"
              />
            </div>
            <div className="form-group">
              <button className="btn btn-primary btn-block" type="submit" onClick={this.handleSubmit}>
                Login
              </button>
            </div>
          </div>
          <a href="#" className="forgot">
            Forgot your email or password?
          </a>
        </form>
      </div>
    );
  }
}

Login.propTypes = {
  login: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  loginReducer: state.loginReducer,
});

export default connect(mapStateToProps, { login: loginAction })(Login);
