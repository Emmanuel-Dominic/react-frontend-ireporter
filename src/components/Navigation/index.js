/* eslint-disable jsx-a11y/alt-text */
/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { Link } from 'react-router-dom';
import 'assets/css/Navigation.css';
import history from 'utils/History';

export default class Navigation extends React.Component {
  handleClick = (event) => {
    event.preventDefault();
    sessionStorage.removeItem('token');
    sessionStorage.setItem('isLoggedIn', false);
    history.push('/login');
  };

  render() {
    const loggedIn = sessionStorage.getItem('isLoggedIn');
    return (
      <nav className="navbar navbar-light navbar-expand-md navbar-fixed-top navigation-clean-button">
        <div className="container">
          <Link to="/" className="navbar-brand">
            <span className="head-text">&#8465;&#8477;&#8455;</span>
            PORT
            <span className="head-text">&#8455;&#8477;</span>
          </Link>
          <button
            type="button"
            data-toggle="collapse"
            data-target="#navcol-1"
            className="navbar-toggler"
          >
            <span className="sr-only">Toggle navigation</span>
            <span className="navbar-toggler-icon" />
          </button>
          <div className="collapse navbar-collapse" id="navcol-1">
            <div className="ml-auto navbar-text actions">
              {
                (loggedIn === 'true')
                  ? (
                    <React.Fragment>
                      <Link to="/home" className="btn btn-light action-buttons">
                        Home
                      </Link>
                      <Link to="/login" className="btn btn-light action-buttons" onClick={this.handleClick}>logout</Link>
                    </React.Fragment>
                  )
                  : (
                    <React.Fragment>
                      <Link to="/login" className="btn btn-light action-buttons">Log In</Link>
                    </React.Fragment>
                  )
              }
            </div>
          </div>
        </div>
      </nav>
    );
  }
}
