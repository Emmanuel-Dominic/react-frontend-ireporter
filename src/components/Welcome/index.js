import React, { Component } from 'react';
import 'assets/scss/Welcome.scss';

class Welcome extends Component {
  render() {
    return (
      <div className="body"><h1 className="welcome">Welcome to Ireporter</h1></div>
    );
  }
}
export default Welcome;
