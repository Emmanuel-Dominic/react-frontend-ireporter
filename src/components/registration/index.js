import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Registration from 'store/actions/registration';
import '../../assets/scss/Signup.scss';

export class Signup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      userName: '',
      firstName: '',
      lastName: '',
      phoneNumber: '',
      password: '',
      emailError: '',
      usernameError: '',
      firstnameError: '',
      lastnameError: '',
      phonenumberError: '',
      passwordError: '',
      confirmpassword: '',
      confirmPasswordError: '',
    };
  }

  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  validateFormData = () => {
    const {
      email,
      userName,
      firstName,
      lastName,
      phoneNumber,
      password,
      confirmpassword,
    } = this.state;

    // eslint-disable-next-line no-useless-escape
    const passwordValidation = new RegExp('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,})');
    // eslint-disable-next-line no-useless-escape
    const usernameValidation = new RegExp('^([a-zA-Z\d]+[-_])*[a-zA-Z\d*]+$');

    this.setState({
      emailError: email.length > 8 ? null : 'Please enter a vaild Email Address',
      usernameError: userName.length > 3 && usernameValidation.test(userName) ? null : 'Username should be 4 characters and cannot be integers, have white spaces or symbol',
      firstnameError: firstName.length > 4 && usernameValidation.test(firstName) ? null : 'firstname should be 4 characters and cannot be integers, have white spaces or symbol',
      lastnameError: lastName.length > 4 && usernameValidation.test(lastName) ? null : 'lastname should be 4 characters and cannot be integers, have white spaces or symbol',
      phonenumberError: phoneNumber.length > 9 ? null : 'Please provide a valid Phonenumber of 10 digits and starting with \'07\'',
      passwordError: passwordValidation.test(password) ? null : 'Password should be 8 to more characters with atleast a number, capital and small letter.',
      confirmPasswordError: confirmpassword === password ? null : 'Passwords don\'t match',
    });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const {
      email, userName, firstName, lastName, phoneNumber, password
    } = this.state;
    const { Register } = this.props;
    const newUser = {
      email, userName, firstName, lastName, phoneNumber, password
    };
    Register(newUser);
    this.setState({
      email: '',
      userName: '',
      firstName: '',
      lastName: '',
      phoneNumber: '',
      password: '',
      confirmpassword: '',
    });
  }

  render() {
    const {
      email,
      userName,
      firstName,
      lastName,
      phoneNumber,
      password,
      confirmpassword,
      emailError,
      usernameError,
      firstnameError,
      lastnameError,
      phonenumberError,
      passwordError,
      confirmPasswordError,
    } = this.state;

    return (
      <div className="register-photo body">
        <div className="form-container container">
          <div className="image-holder" />
          <form>
            <h2 className="text-center">
              <strong className="head-text">Create </strong>
            an account
            </h2>
            <div>
              <input
                id="email"
                type="email"
                name="email"
                value={email}
                onChange={this.handleChange}
                onKeyUp={this.validateFormData}
                placeholder="Email"
                className="form-control"
                required
              />
              <p className="error">{emailError}</p>
              <input
                id="username"
                type="text"
                name="userName"
                value={userName}
                onChange={this.handleChange}
                onKeyUp={this.validateFormData}
                placeholder="Username"
                className="form-control"
                required
              />
              <p className="error">{usernameError}</p>
              <input
                id="firstname"
                type="text"
                name="firstName"
                value={firstName}
                onChange={this.handleChange}
                onKeyUp={this.validateFormData}
                placeholder="firstName"
                className="form-control"
                required
              />
              <p className="error">{firstnameError}</p>
              <input
                id="lastname"
                type="text"
                name="lastName"
                value={lastName}
                onChange={this.handleChange}
                onKeyUp={this.validateFormData}
                placeholder="lastName"
                className="form-control"
                required
              />
              <p className="error">{lastnameError}</p>
              <input
                id="phonenumber"
                type="number"
                name="phoneNumber"
                value={phoneNumber}
                onChange={this.handleChange}
                onKeyUp={this.validateFormData}
                placeholder="phoneNumber"
                className="form-control"
                required
              />
              <p className="error">{phonenumberError}</p>
              <input
                id="password"
                type="password"
                name="password"
                value={password}
                onChange={this.handleChange}
                onKeyUp={this.validateFormData}
                placeholder="Password"
                className="form-control"
                required
              />
              <p className="error">{passwordError}</p>
              <input
                id="confirmpassword"
                type="password"
                name="confirmpassword"
                value={confirmpassword}
                onChange={this.handleChange}
                onKeyUp={this.validateFormData}
                placeholder="Confirmpassword"
                className="form-control"
                required
              />
              <p className="error">{confirmPasswordError}</p>
            </div>
            <div className="form-group">
              <Link to="/" className="already">Changed your mind? Cancel registration.</Link>
            </div>
            <button type="submit" onClick={this.handleSubmit} className="btn btn-primary btn-block">Signup</button>
            <Link to="/login" className="already">You already have an account? Login here.</Link>
          </form>
        </div>
      </div>
    );
  }
}

Signup.propTypes = {
  Register: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  user: state.user,
});

export default connect(mapStateToProps, { Register: Registration })(Signup);
