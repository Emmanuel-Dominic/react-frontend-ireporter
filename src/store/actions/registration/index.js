import axios from 'axios';
import { RegistrationConstants } from 'store/actions/actionTypes';
import { toastSuccess, toastFailure } from 'utils/Toast';
import history from 'utils/History';


export const Registration = newUser => (dispatch) => {
  dispatch({
    type: RegistrationConstants.REGISTER_REQUEST,
  });
  return axios
    .post('http://127.0.0.1:5000/api/v3/auth/signup', newUser, {
      headers: {
        'content-type': 'application/json',
      },
    })
    .then((response) => {
      dispatch({
        type: RegistrationConstants.REGISTER_SUCCESS,
        payload: response.data.message,
      });
      toastSuccess(`Account created and you are ${response.data.message}`, 'A');
      history.push('/login');
    })
    .catch((error) => {
      if (error.response.status === 400) {
        dispatch({
          type: RegistrationConstants.REGISTER_FAILURE,
          payload: error.response.data.error.message,
        });
        toastFailure(`${error.response.data.error.message}`, 'A');
      } else if (error.response.status === 406) {
        dispatch({
          type: RegistrationConstants.REGISTER_FAILURE,
          payload: error.response.data.error,
        });
        toastFailure('All fields are required', 'A');
      }
    });
};

export default Registration;
