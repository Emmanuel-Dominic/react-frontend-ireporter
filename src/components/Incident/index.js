/* eslint-disable no-underscore-dangle */
import { connect } from 'react-redux';
import { ToastContainer } from 'react-toastify';
import PropTypes from 'prop-types';
import React from 'react';

import '../../assets/scss/Home.scss';
import { IncidentsAction } from 'store/actions/Incident';

export class ViewIncident extends React.Component {
  componentDidMount() {
    const { viewIncidents } = this.props;
    viewIncidents('intervention');
  }

  render() {
    const {
      incidents,
    } = this.props;
    return (
      <React.Fragment>
        <ToastContainer />
        <div className="incident-body">
          {incidents.map(incident => (
            <div className="container card card-body" key={incident.incident_id}>
              <h4 className="col-md-10 title">{incident.title}</h4>
              <div className="row">
                <p className="col-md-2">{incident.user_name}</p>
                <p className="col-md-2">{incident.status_}</p>
              </div>
              <div className="row">
                <p className="col-md-2">{incident.latitude}</p>
                <p className="col-md-2">{incident.longtitude}</p>
              </div>
              <p>{incident.comment}</p>
            </div>
          ))}
        </div>
      </React.Fragment>
    );
  }
}

export const mapStateToProps = ({ incidents }) => ({
  incidents: incidents.incidents,
});

ViewIncident.propTypes = {
  viewIncidents: PropTypes.func,
}.isRequired;

ViewIncident.defaultProps = {
  viewIncidents: () => {},
};

export default connect(
  mapStateToProps,
  { viewIncidents: IncidentsAction },
)(ViewIncident);
