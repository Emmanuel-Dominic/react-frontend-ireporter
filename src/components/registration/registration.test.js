import React from 'react';
import { shallow } from 'enzyme';
import { Signup, mapStateToProps } from 'components/registration';


function renderSignup(args) {
  const defaultProps = {
    actions: { Register: jest.fn() },
    user: {},
    Register: jest.fn(),
  };
  const props = { ...defaultProps, ...args };
  return shallow(<Signup {...props} />);
}

describe('registration page', () => {
  const wrapper = renderSignup();
  const wrapperInst = wrapper.instance();

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle input change', () => {
    const event = {
      target: {
        name: 'userName',
        value: 'emma',
      },
    };
    wrapperInst.handleChange(event);
    expect(wrapperInst.state.userName).toBe('emma');
  });

  it('should handle submit and clear form after', () => {
    wrapper.setState({
      userName: 'emma',
      firstName: 'emma',
      lastName: 'emma',
      phoneNumber: '07000000000',
      email: 'email@email.com',
      password: 'password',
      confirmpassword: 'confirmpassword',
    });

    expect(wrapperInst.state.userName).toBe('emma');
    expect(wrapperInst.state.firstName).toBe('emma');
    expect(wrapperInst.state.lastName).toBe('emma');
    expect(wrapperInst.state.phoneNumber).toBe('07000000000');
    expect(wrapperInst.state.email).toBe('email@email.com');
    expect(wrapperInst.state.password).toBe('password');
    expect(wrapperInst.state.confirmpassword).toBe('confirmpassword');

    const event = {
      Register: jest.fn(),
      target: {
        type: 'submit',
      },
      preventDefault: jest.fn(),
    };
    wrapperInst.handleSubmit(event);
    expect(wrapperInst.state.userName).toBe('');
    expect(wrapperInst.state.firstName).toBe('');
    expect(wrapperInst.state.lastName).toBe('');
    expect(wrapperInst.state.phoneNumber).toBe('');
    expect(wrapperInst.state.email).toBe('');
    expect(wrapperInst.state.password).toBe('');
    expect(wrapperInst.state.confirmpassword).toBe('');
  });

  it('should map tostate to props', () => {
    const mockedState = {
      user: {
        registering: true,
      },
    };
    const state = mapStateToProps(mockedState);
    expect(state.user.registering).toBe(true);
  });
});
