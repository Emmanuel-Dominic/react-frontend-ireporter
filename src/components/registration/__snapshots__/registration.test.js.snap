// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registration page should match the snapshot 1`] = `
<div
  className="register-photo body"
>
  <div
    className="form-container container"
  >
    <div
      className="image-holder"
    />
    <form>
      <h2
        className="text-center"
      >
        <strong
          className="head-text"
        >
          Create 
        </strong>
        an account
      </h2>
      <div>
        <input
          className="form-control"
          id="email"
          name="email"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
        <p
          className="error"
        />
        <input
          className="form-control"
          id="username"
          name="userName"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
        <p
          className="error"
        />
        <input
          className="form-control"
          id="firstname"
          name="firstName"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="firstName"
          required={true}
          type="text"
          value=""
        />
        <p
          className="error"
        />
        <input
          className="form-control"
          id="lastname"
          name="lastName"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="lastName"
          required={true}
          type="text"
          value=""
        />
        <p
          className="error"
        />
        <input
          className="form-control"
          id="phonenumber"
          name="phoneNumber"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="phoneNumber"
          required={true}
          type="number"
          value=""
        />
        <p
          className="error"
        />
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <p
          className="error"
        />
        <input
          className="form-control"
          id="confirmpassword"
          name="confirmpassword"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Confirmpassword"
          required={true}
          type="password"
          value=""
        />
        <p
          className="error"
        />
      </div>
      <div
        className="form-group"
      >
        <Link
          className="already"
          to="/"
        >
          Changed your mind? Cancel registration.
        </Link>
      </div>
      <button
        className="btn btn-primary btn-block"
        onClick={[Function]}
        type="submit"
      >
        Signup
      </button>
      <Link
        className="already"
        to="/login"
      >
        You already have an account? Login here.
      </Link>
    </form>
  </div>
</div>
`;
